### PR TITLE
feat(*): token cleanup [KHCP-8461]

### PR DESCRIPTION
**Pull request by portikM**
# Summary

Addresses: https://konghq.atlassian.net/browse/KHCP-8461

Cleanup token misusage, utility classes and old CSS variables (apart from utility classes in docs files).

## PR Checklist

* [ ] **Conventional Commits** all commits follow the conventional commit standards [outlined in the main README](https://github.com/Kong/kongponents#committing-changes).
* [ ] **Functional:** all changes do not break existing APIs and if so, bump major version.
* [ ] **Tests pass:** check the output of yarn test
* [ ] **Naming:** the files and the method and prop variables use the same naming conventions as other Kongponents
* [ ] **Framework style:** abides by the essential rules in Vue's style guide
* [ ] **Cleanliness:** does not have formatting issues, unused code (e.g., console.logs, debugger), or leftover comments
* [ ] **Docs:** includes a technically accurate README, uses JSDOC where appropriate

### CODE REVIEW

File: .eslintrc.cjs [hunk 1]
```diff
@@ -34,6 +36,8 @@ module.exports = {
     'vue/attributes-order': ['error', {
       alphabetical: true,
     }],
+    // Disallow Kongponents utility classes
+    'vue/no-restricted-class': ['error', ...deprecatedUtilityClasses],
```
Comment (adamdehaven) [inline .eslintrc.cjs:40]: Added this to prevent more usage of deprecated utility classes

File: .stylelintrc.cjs [hunk 2]
```diff
@@ -14,17 +14,31 @@ module.exports = {
...
         ],
+        // TODO: Enable once all deprecated CSS custom properties have been removed in breaking version.
+        //
+        // Only allow @kong/design-tokens CSS custom properties
+        // 'custom-property-pattern': [
+        //   "^(kui).+$",
+        //   {
+        //     message: "Expected custom property \"%s\" to be sourced from @kong/design-tokens with prefix '--kui-'",
+        //   }
+        // ],
```
Comment (adamdehaven) [inline .stylelintrc.cjs:25]: I'm leaving this disabled for now until after we remove utility css custom properties in the breaking release

File: deprecated-classes.cjs [hunk 3]
```diff
@@ -0,0 +1,288 @@
+// List of disallowed utility classes that are exported from Kongponents
+// These classes **must not** be used in this repository going forward
+module.exports = [
```
Comment (adamdehaven) [inline deprecated-classes.cjs:3]: This is the generated class list; should not be manually updated.

File: src/components/KCatalog/KCatalogItem.vue [hunk 4]
```diff
@@ -69,8 +69,8 @@ const handleCardClick = (evt: Event, item: CatalogItem): void => {
   overflow: hidden;
 
   .multi-line-truncate {
-    line-height: var(--lh, var(--kui-line-height-30, $kui-line-height-30));
-    max-height: calc(var(--lh, var(--kui-line-height-30, $kui-line-height-30)) * var(--max-lines, 5));
+    line-height: var(--lh, 20px);
+    max-height: calc(var(--lh, 20px) * var(--max-lines, 5));
```
Comment (adamdehaven) [inline src/components/KCatalog/KCatalogItem.vue:73]: question: what are these css custom properties...?

Comment (adamdehaven) [reply src/components/KCatalog/KCatalogItem.vue:73]: They don't follow the standard theming pattern, so I'd say remove them. There are also some in KPopover (or KTooltip, can't remember) -- are they created by us? If yes, let's remove if it doesn't break anything. 

If they are from the popper.js or something, then leave in place

File: src/components/KAlert/KAlert.vue [hunk 5]
```diff
@@ -327,7 +327,7 @@ export const appearances: AlertAppearanceRecord = {
     }
 
     p:last-of-type {
-      margin-bottom: var(--kui-space-0, $kui-space-0);
+      margin-bottom: 0;
```
[the remark this answers is not in the document]
Comment (portikM) [reply src/components/KAlert/KAlert.vue:330]: I'm removing misused tokens (space tokens used in positioning properties). This `margin-bottom` got caught in the crossfire by accident. Fixed now.

File: src/components/KCatalog/KCatalogItem.vue (hunk 4 above)
Comment (portikM) [reply src/components/KCatalog/KCatalogItem.vue]: Removed these variables. Both KPopover and KTooltip follow the standard theming pattern - but if we find other Kongponents using non-standard variables we can remove them later.

**Review by adamdehaven — DISMISSED** [review]
lgtm

**Review by adamdehaven — APPROVED** [review]

**Pull request merged**